src/main/java/Duke.java

```diff
@@ -1,10 +1,225 @@
+        // Return the <subcommand>
+        // For example, the deadline subcommand: "by Friday, 13th"
+        // Should return "by"
+        String[] words = subcommand.split(" ");
+        return words[0];
+    }
+
+    public static String extractTextFromSubcommand(String subcommand) {
+        // Subcommands have format <subcommand> <text_to_extract>
+        // Return the <subcommand> and <text_to_extract> in a string array
+        // For example, the deadline subcommand: "by Friday, 13th"
+        // Should return ["by" , "Friday, 13th"]
+        String[] words = subcommand.split(" ");
+        String subcommandText = "";
+        for (int i = 1; i < words.length; ++i) {
+            subcommandText = subcommandText.concat(words[i]);
+            if (i != words.length - 1) {
+                subcommandText = subcommandText.concat(" ");
+            }
+        }
+        return subcommandText;
+    }
+
+    public static Task createTask(HashMap<String, String> commandMap, String taskType) {
+        // Start from 1 as cop
+        Task newTask = null;
+        if (taskType.equals("todo")) {
+            // To-do task
+            String description = commandMap.get("todo");
+            newTask = new Todo(description);
+
+        } else if (taskType.equals("deadline")) {
+            // Guard clauses to check for validity
+            if (commandMap.size() < 2) {
+                System.out.println("Not enough arguments!");
+                return newTask;
+            }
+            if (!commandMap.containsKey("by")) {
+                System.out.println("Missing --by argument!");
+                return newTask;
+            }
+            // Deadline task
+            String description = commandMap.get("deadline");
+            String deadline = commandMap.get("by");
+            newTask = new Deadline(description, deadline);
+        } else if (taskType.equals("event")) {
+
+            if (commandMap.size() < 3) {
+                System.out.println("Not enough arguments!");
+                return newTask;
+            }
+            if (!commandMap.containsKey("from")) {
+                System.out.println("Missing --from argument!");
+                return newTask;
+            }
+            if (!commandMap.containsKey("to")) {
+                System.out.println("Missing --to argument!");
+                return newTask;
+            }
+            // Event task
+            String description = commandMap.get("event");
+            String from = commandMap.get("from");
+            String to = commandMap.get("to");
+            newTask = new Event(description, from, to);
+        }
+        return newTask;
+    }
+
+    public static void handleCommand(String command) {
+        // Update the keepAlive flag
+        String[] commandList = splitIntoSubcommands(command);
+        String mainCommand = extractCommandFromSubcommand(commandList[0]);
+        switch (mainCommand) {
+        case "Bye":
```

Comment:
good that the methods and variables are named properly that clearly shows the intent of it